contracts/ETHExpirationCollector.sol

```diff
@@ -0,0 +1,366 @@
+     * @param id The license to make a payment for
+     */
+    function makePayment(uint256 id) external payable whenNotPaused {
+        uint256 currentContributionRate = accountant.contributionRates(id);
+
+        // Update expiration
+        _updateExpiration(id, currentContributionRate, msg.value, 0);
+
+        // Transfer payment to receiver
+        _asyncTransfer(receiver, msg.value);
+
+        emit PaymentMade(id, msg.value);
+    }
+
+    /**
+     * @notice Set the contribution rate for a license and optionally make a payment
+     * @param id The license to make a payment for
+     * @param newContributionRate The new contribution rate for the license
+     * @custom:requires MODIFY_CONTRIBUTION_ROLE or sender is license owner
+     */
+    function setContributionRate(uint256 id, uint256 newContributionRate)
+        external
+        payable
+        whenNotPaused
+    {
+        require(
+            hasRole(MODIFY_CONTRIBUTION_ROLE, msg.sender) ||
+                license.ownerOf(id) == msg.sender ||
+                license.isApprovedForAll(license.ownerOf(id), msg.sender) ||
+                license.getApproved(id) == msg.sender,
+            "Caller does not have permission"
+        );
+        require(
+            newContributionRate >= minContributionRate,
+            "Contribution rate must be greater than minimum"
+        );
+
+        // Update expiration
+        _updateExpiration(id, newContributionRate, msg.value, 0);
+
+        // Update contribution rate in Accountant
+        accountant.setContributionRate(id, newContributionRate);
+    }
+
+    /**
+     * @notice Migrate all funds from one license to another and clear the from license
+     * @param fromId The license to migrate from and clear
+     * @param toId The license to migrate to
+     * @custom:requires MODIFY_FUNDS_ROLE
+     */
+    function migrateFunds(
+        uint256 fromId,
+        uint256 toId,
+        uint256 toContributionRate
+    ) external payable onlyRole(MODIFY_FUNDS_ROLE) {
+        require(
+            toContributionRate >= minContributionRate,
+            "Contribution rate must be greater than minimum"
+        );
+
+        uint256 fromNetworkFeeBalance = _calculateNetworkFeeBalance(fromId);
+        _updateExpiration(
+            toId,
+            toContributionRate,
+            fromNetworkFeeBalance + msg.value,
+            0
+        );
+
+        // Clear from license
+        licenseExpirationTimestamps[fromId] = 0;
+        accountant.setContributionRate(fromId, 0);
+
+        // Update contribution rate in Accountant
+        accountant.setContributionRate(toId, toContributionRate);
+    }
+
+    /**
+     * @notice Move funds from one license to another
+     * @param fromId The license to move from
+     * @param toId The license to move to
+     * @param amount The amount of funds to move
+     * @custom:requires MODIFY_FUNDS_ROLE
+     */
+    function moveFunds(
```

Comment:
what circumstance would this method be used for?

Reply:
This will mainly be used when splitting a parcel. The existing parcel will need to "move" funds to the new parcel.